domain/application/service/application.go

```diff
@@ -674,6 +680,23 @@ func makeCloudContainerArg(unitName coreunit.Name, cloudContainer CloudContainer
 	return result
 }
 
+// makeResourcesArgs creates a slice of AddApplicationResourceArg from ResolvedResources.
+// Returns an error if any of the ResolvedResources are invalid.
+func makeResourcesArgs(resolvedResources ResolvedResources) ([]application.AddApplicationResourceArg, error) {
```

Comment:
todo: break the method into 2 pieces: validation and transformation/encoding.

The validation should be done in validateCreateApplicationParams().